### PR TITLE
Add item map and navigation by foot

### PR DESCRIPTION
**Pull request by omarelmalki**
Display items that have a last location saved on a map.
Each item is clickable and displays its name and 2 buttons when clicked.
The buttons "Walk" and "Drive" open a new activity and start a navigation from the device's current location to the object's location with a voice guide, the ETA etc..
When a user sends a location on the chat, clicking on it displays the item on the map in a similar way as if the map was opened at first and the same interactions/navigations are possible.

For testing, I have an issue with the firebase emulator and cannot run tests locally.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -236,17 +241,17 @@ connectedCheck {
 }
 
 task accessToken {
-    def tokenFile = new File("${projectDir}/src/main/res/values/developer-config.xml")
+    def tokenFile = new File("${projectDir}/src/main/res/values/developer-config.xml");
```
Comment (DeepGreen1) [inline app/build.gradle:244]: I think that until now, we didn't had to put ";" at the end of lines (similar as Kotlin code). Out of curiosity, why did you change it?

Comment (DeepGreen1) [reply app/build.gradle:244]: The same remark applies for the changes of this file

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -67,11 +59,10 @@
             android:name="com.google.firebase.messaging.default_notification_color"
             android:resource="@color/colorAccent" />
 
-        <activity
-            android:name=".ui.MapsActivity"
-            android:exported="false"
-            android:label="@string/title_activity_maps" />
-
+<!--        <activity-->
+<!--            android:name=".ui.MapsActivity"-->
```
Comment (DeepGreen1) [inline app/src/main/AndroidManifest.xml:63]: If we decide to never use the MapsActivity again, it can be good to simply delete this part of code in order to keep clean code.

File: app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt [hunk 3]
```diff
@@ -119,6 +120,11 @@ class ChatActivity : AppCompatActivity() {
                     adapter.getItemViewType(position) == ChatMessagesListAdapter.MessageType.TYPE_LOCATION_LEFT.ordinal
                 ) {
                     val myIntent = Intent(this@ChatActivity, MapBoxActivity::class.java)
+                    val message = adapter.getItem(position) as LocationMessage
+                    println(message.location.toAndroidLocation().longitude)
```
Comment (DeepGreen1) [inline app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt:124]: Those two println() were certainly used for testing. We discussed the problems that we are all facing with Cirrus so it's ok if you keep it for now but don't forget to remove it afterwards.

**Review by DeepGreen1 — APPROVED** [review]
Nice work! The feature is working on a physical device and is pleasant to use, that is really cool!
I thought about something that could improve the user experience: when the location is on another continent, we cannot walk/drive there; it would be nice to have a message telling us that.

The code coverage is low and we discussed that you will be doing testing in order to get a better coverage during the next sprint. The issue comes from Cirrus/Firebase emulator and we are all more or less affected by this situation; in consequence we agreed to do so.

Anyhow, I let you check the few comments in the code and again, great job!

File: app/src/androidTest/java/com/github/brugapp/brug/ChatActivityTest.kt [hunk 4]
```diff
@@ -383,34 +383,34 @@ class ChatActivityTest {
...
+//                    click()
+//                )
+//            )
+//
+//            val expectedIntent: Matcher<Intent> = anyOf(
+//                hasComponent(MapBoxActivity::class.java.name)
+//            )
+//            intended(expectedIntent)
+//        }
+//    }
```
Comment (Mouniraki) [inline app/src/androidTest/java/com/github/brugapp/brug/ChatActivityTest.kt:413]: Is there any reason why you commented the whole test ? Consider maybe putting a little informative comment on top of it.

File: app/src/androidTest/java/com/github/brugapp/brug/MapsActivityTest.kt [hunk 5]
```diff
@@ -1,56 +1,55 @@
...
+//    fun navigateButtonOpensGoogleMaps(){
+//        val intent = Intent(ApplicationProvider.getApplicationContext(), MapsActivity::class.java)
+//        ActivityScenario.launch<MapsActivity>(intent).use {
+//            val navigateButton = Espresso.onView(ViewMatchers.withId(R.id.navigateButton))
+//            navigateButton.perform(ViewActions.click())
+//            Intents.intended(toPackage("com.google.android.apps.maps"))
+//        }
+//    }
+//
+//}
```
Comment (Mouniraki) [inline app/src/androidTest/java/com/github/brugapp/brug/MapsActivityTest.kt:55]: Same here, why the class is entirely commented ?

File: app/src/main/java/com/github/brugapp/brug/model/LonLatCoordinates.kt [hunk 6]
```diff
@@ -0,0 +1,3 @@
+package com.github.brugapp.brug.model
+
+class LonLatCoordinates(val lon: Double, val lat: Double)
```
Comment (Mouniraki) [inline app/src/main/java/com/github/brugapp/brug/model/LonLatCoordinates.kt:3]: Why did you create a new class handling the location objects ? There is already a LocationService class specially made for that purpose in our project.

File: app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt (hunk 3 above)
Comment (omarelmalki) [reply app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt:124]: Thanks for the catch!

File: app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (omarelmalki) [reply app/src/main/AndroidManifest.xml:63]: I'm commenting everything related to it as it can still be used but not needed.

File: app/src/main/java/com/github/brugapp/brug/ui/MapsActivity.kt [hunk 7]
```diff
@@ -1,87 +1,87 @@
...
+//    override fun onMapReady(googleMap: GoogleMap) {
+//        mMap = googleMap
+//
+//        // Add a marker in Destination and move the camera
+//        val marker = LatLng(viewModel.getDestinationLat(), viewModel.getDestinationLon())
+//        mMap.addMarker(MarkerOptions().position(marker).title(viewModel.getDestinationName()))
+//        mMap.moveCamera(CameraUpdateFactory.newLatLng(marker))
+//    }
+//
+//}
```
Comment (Mouniraki) [inline app/src/main/java/com/github/brugapp/brug/ui/MapsActivity.kt:87]: Same as previous comments on big chunk of commented code.

File: app/build.gradle (hunk 1 above)
Comment (omarelmalki) [reply app/build.gradle:244]: That happened when I merged the main branch into mine, I didn't pay attention to that semi-colon.

**Review by Mouniraki — APPROVED** [review]
Overall a pretty decent job, as I understood you had trouble testing everything on your computer due to the fact that Firebase Emulator is not working properly on your end. We will have to fix this ASAP in order for you to be able to test more your code, since the coverage is a bit low. Consider my few comments before merging.

File: app/src/androidTest/java/com/github/brugapp/brug/ChatActivityTest.kt (hunk 4 above)
Comment (omarelmalki) [reply app/src/androidTest/java/com/github/brugapp/brug/ChatActivityTest.kt:413]: Yes, this test consistently failed on Cirrus CI after I merged the main branch into mine and I had to comment it to run it.

File: app/src/main/java/com/github/brugapp/brug/model/LonLatCoordinates.kt (hunk 6 above)
Comment (omarelmalki) [reply app/src/main/java/com/github/brugapp/brug/model/LonLatCoordinates.kt:3]: This class was created a while back. I will use LocationService instead, you are right.

File: app/src/androidTest/java/com/github/brugapp/brug/MapsActivityTest.kt (hunk 5 above)
Comment (omarelmalki) [reply app/src/androidTest/java/com/github/brugapp/brug/MapsActivityTest.kt:55]: This class is not used anymore as we opted for MapBox instead of Google Maps. I decided to comment it and its test class before deleting it once we discuss the next steps.

File: app/src/main/java/com/github/brugapp/brug/ui/MapsActivity.kt (hunk 7 above)
Comment (omarelmalki) [reply app/src/main/java/com/github/brugapp/brug/ui/MapsActivity.kt:87]: This class is not used anymore as we opted for MapBox instead of Google Maps. I decided to comment it and its test class before deleting it once we discuss the next steps.

File: app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (omarelmalki) [reply app/src/main/AndroidManifest.xml]: Fixed in my last commit.

File: app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt (hunk 3 above)
Comment (omarelmalki) [reply app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt]: Fixed in my last commit.

File: app/src/androidTest/java/com/github/brugapp/brug/ChatActivityTest.kt (hunk 4 above)
Comment (omarelmalki) [reply app/src/androidTest/java/com/github/brugapp/brug/ChatActivityTest.kt:414]: Fixed in my last commit.

File: app/src/androidTest/java/com/github/brugapp/brug/MapsActivityTest.kt (hunk 5 above)
Comment (omarelmalki) [reply app/src/androidTest/java/com/github/brugapp/brug/MapsActivityTest.kt]: Fixed in my last commit.

File: app/src/main/java/com/github/brugapp/brug/model/LonLatCoordinates.kt (hunk 6 above)
Comment (omarelmalki) [reply app/src/main/java/com/github/brugapp/brug/model/LonLatCoordinates.kt]: Fixed in my last commit.

File: app/src/main/java/com/github/brugapp/brug/ui/MapsActivity.kt (hunk 7 above)
Comment (omarelmalki) [reply app/src/main/java/com/github/brugapp/brug/ui/MapsActivity.kt]: Fixed in my last commit.
